autora/synthetic/inventory.py

```diff
@@ -73,11 +99,13 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from typing import Any, Callable, Dict, Optional, Protocol
+from functools import singledispatch
+from typing import Any, Callable, Dict, Optional, Protocol, runtime_checkable
 
 from autora.variable import VariableCollection
 
 
+@runtime_checkable
```

Comment:
What does this decorator do?

Reply:
That means that you can do an `isinstance(x, _SyntheticExperimentClosure)` call and check whether `x` fulfills the Protocol.

Reply:
Oh cool! Thanks!